src/commands/diff.ts

```diff
@@ -80,23 +83,35 @@ export default class Diff extends Command {
 
     cli.action.stop();
 
-    if (version) {
-      await this.displayCompareResult(version, flags.open);
+    if (diff) {
+      /* Flags format has a default value, so it's always defined. But
+       * oclif types can"t detect it */
+      /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+      await this.displayCompareResult(diff, flags.format!, flags.open);
     } else {
       await cli.log('No changes detected.');
     }
 
     return;
   }
 
-  async displayCompareResult(version: VersionResponse, open: boolean): Promise<void> {
-    if (version && version.diff_summary) {
-      await cli.log(version.diff_summary);
-      if (open && version.diff_public_url) {
-        await cli.open(version.diff_public_url);
-      }
+  async displayCompareResult(
+    result: WithDiff,
+    format: string,
+    open: boolean,
+  ): Promise<void> {
+    if (format == 'text' && result && result.diff_summary) {
```

Comment:
You should be able to use this shorthand:
```suggestion
    if (format == 'text' && result?.diff_summary) {
```
Or maybe test for result first then test for format?
```suggestion
    if (result) {
      if (format == 'text' && result.diff_summary) {
        ...
      }
    } else {
      await cli.log('No structural changes detected');
```

Reply:
I'll favor the second suggestion is sound better indeed. Thanks!

Reply:
Wait, I've got typing in typescript to avoid this can of thing. And the function signature doesn't allow an undefined `result` variable.

Thus I can simply remove this condition!

Thanks for pointing this out.

Reply:
It looks way better indeed!